library/core/src/main/java/com/google/android/exoplayer2/source/ads/AdsMediaSource.java

```diff
@@ -263,7 +263,7 @@ protected void onChildSourceInfoRefreshed(
       checkNotNull(adMediaSourceHolders[adGroupIndex][adIndexInAdGroup])
           .handleSourceInfoRefresh(newTimeline);
     } else {
-      Assertions.checkArgument(newTimeline.getPeriodCount() == 1);
+      // Assertions.checkArgument(newTimeline.getPeriodCount() == 1);
```

Comment:
We cannot simply comment out the single-period check (two places to check, we need to comment [this code](https://github.com/DiceTechnology/ExoPlayer/blob/0179cc1de08d75ad99b50b96caa1d383d7f34bdc/library/core/src/main/java/com/google/android/exoplayer2/source/ads/AdsMediaSource.java#L490)), because this assumption is used in many places in the `AdMediaSource` class. 
```
Object periodUid = timeline.getUidOfPeriod(/* periodIndex= */ 0);

timeline.getPeriod(/* periodIndex= */ 0, period).getDurationUs();
```
For multi-periods, many codes are logically incorrect, we need to modify these codes, which is not easy.

In addition, timeline is widely used by ExoPlayerImplInternal/PlaybackInfo in many places. It is the core object of ExoPlayer and mainly affects seek. It has a relatively large impact on mid-roll/post-roll.

This requires a lot of testing to find the relevant problems, but from the code level, it is not appropriate to directly comment out the single-period check without modifying the relevant code (which may be more and not so easy to modify).